Sources/OpenCombineFoundation/Publishers+Timer.swift

```diff
@@ -0,0 +1,345 @@
+        }
+
+        func receive(subscription: Subscription) {
+          lock.lock()
+          let subscribers = self.subscribers
+          lock.unlock()
+
+          for sub in subscribers {
+            sub.receive(subscription: subscription)
+          }
+        }
+
+        func receive(_ value: Input) -> Subscribers.Demand {
+          var resultingDemand: Subscribers.Demand = .max(0)
+          lock.lock()
+          let subscribers = self.subscribers
+          let isConnected = _lockedIsConnected
+          lock.unlock()
+
+          guard isConnected else { return .none }
+
+          for sub in subscribers {
+            resultingDemand += sub.receive(value)
+          }
+          return resultingDemand
+        }
+
+        func receive(completion: Subscribers.Completion<Failure>) {
+          lock.lock()
+          let subscribers = self.subscribers
+          lock.unlock()
+
+          for sub in subscribers {
+            sub.receive(completion: completion)
+          }
+        }
+
+        func request(_ demand: Subscribers.Demand) {
+          lock.lock()
+          // Inner will always be non-nil
+          let inner = self.inner!
+          lock.unlock()
+
+          inner.request(demand)
+        }
+
+        func cancel() {
+          lock.lock()
+          // Inner will always be non-nil
+          let inner = self.inner!
+          _lockedIsConnected = false
+          subscribers = []
+          lock.unlock()
+
+          inner.cancel()
+        }
+      }
+
+      public func receive<Sub: Subscriber>(subscriber: Sub)
+      where Failure == Sub.Failure, Output == Sub.Input {
+        routingSubscription.addSubscriber(subscriber)
+      }
+
+      public func connect() -> Cancellable {
+        routingSubscription.isConnected = true
+        return routingSubscription
+      }
+
+      private typealias Parent = TimerPublisher
+      private final class Inner<Downstream: Subscriber>: NSObject, Subscription,
+        CustomReflectable, CustomPlaygroundDisplayConvertible
+        where
+        Downstream.Input == Date,
+        Downstream.Failure == Never {
+        private lazy var timer: Timer? = {
+          let t = Timer(
+            timeInterval: parent?.interval ?? 0,
+            repeats: true
+          ) { [weak self] _ in self?.timerFired() }
+
+          t.tolerance = parent?.tolerance ?? 0
+
+          return t
+        }()
+
+        private let lock: Lock
+        private var downstream: Downstream?
+        private var parent: Parent?
+        private var started: Bool
+        private var demand: Subscribers.Demand
+
+        override var description: String { return "Timer" }
+        var customMirror: Mirror {
+          lock.lock()
+          defer { lock.unlock() }
+          return Mirror(self, children: [
+            "downstream": downstream as Any,
+            "interval": parent?.interval as Any,
+            "tolerance": parent?.tolerance as Any,
+          ])
```

Comment:
Just to be consistent with the rest of our code, where we use arrays instead of dictionaries:
```suggestion
            let children: [Mirror.Child] = [
                ("downstream", downstream as Any),
                ("interval", parent?.interval as Any),
                ("tolerance", parent?.tolerance as Any),
            ]
            return Mirror(self, children: children)
```